src/odemis/acq/stream/_projection.py

```diff
@@ -218,6 +223,437 @@ def _updateImage(self):
+
+                    # TODO allow variable input size? Calc based on raw data?
+                    size = (180, 720)  # (2*90, 2*360) defines the resolution of the displayed image
+
+                    for key in data_raw_bg_corr.keys():
+                        data_raw_bg_corr[key] = angleres.AngleResolved2Rectangular(data_raw_bg_corr[key], size)
+                    polarimetry_dict_raw[ebeam_pos] = data_raw_bg_corr  # dict containing the bg corrected raw data
+
+                    # Calculate the polarimetry results:
+                    # Dictionary keys are (ebeam pos x, ebeam pos y, optional: polarimetry key).
+                    # Note: Takes about 0.25 sec to calc all polarimetry results for one ebeam pos
+                    # and tested on an image of size (256, 1024)
+                    polarimetry_dict_raw.update(
+                        arpolvis.calcPolarimetry(polarimetry_dict_raw, ebeam_pos, size, wl))
+
+                # Convert the data back to polar representation for GUI display.
+                # Note: Takes 0.24 sec to convert one image for display and tested on an image of size (256, 1024)
+                # TODO invalid division check if only bg  # norm so far
+                # TODO move to update image?? as this is the only code we need to reuse
+                #  whenever the legend changes while having the same ebeam pos selected
+                if pos[-1] in ["Ephi", "Etheta", "Ex", "Ey", "Ez"]:  # TODO: hardcoded for now
+                    data = numpy.abs(polarimetry_dict_raw[pos])
+                    order = numpy.log10(data.max())
+                    try:
+                        plotorder = numpy.int(order)
+                    except OverflowError:
+                        # Note: Error is raised, when bg image and image are the same
+                        # (-> cannot convert float infinity to integer)
+                        # TODO is there a better way to handle this problem?
+                        plotorder = 0
+                    polarimetry_dict_raw[pos] = data / 10 ** plotorder
+                    colormap = 2
+                else:
+                    colormap = 1
+                # TODO this gives an e.g. Index/Key error in the logging when failing. why? this code is in a try and except...
+                stokes_data = \
+                    angleres.Rectangular2Polar(polarimetry_dict_raw[pos], colormap=colormap)
+
+                new_md = self._find_metadata(stokes_data.metadata)
+                new_md[model.MD_DIMS] = stokes_data.metadata[model.MD_DIMS] # YXC
+                polarimetry_dict[pos] = model.DataArray(stokes_data, new_md)
+
+                # return an array
+                stokes_data = polarimetry_dict[pos]
+
+            except Exception:
+                logging.exception("Failed to calculate polarimetry results.")
+                return numpy.zeros((1034, 1034, 3))  # return a black image as fallback
```

Comment:
If you have to return an image, at least you could return a smaller one ;-) Probably 10,10,3 is more than enough. Alternatively, you could try to return `None` and handle it properly in the callers.

Reply:
I tried to do it the same way as for the raw ar...I like a black image to be displayed so the user sees not the old image and thinks by accident that this is already the updated image.